src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,186 @@
+		{
+			textStyleClassList.Add((namedTypeSymbol.Name, "public", namedTypeSymbol.ContainingNamespace.ToDisplayString()));
+		}
+
+		// Collect All Classes in User Library that Implement ITextStyle
+		foreach (var classDeclarationSyntax in userGeneratedClassesProvider)
+		{
+			var declarationSymbol = compilation.GetSymbol<INamedTypeSymbol>(classDeclarationSyntax);
+			if (declarationSymbol is null)
+			{
+				var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidClassDeclarationSyntax, Location.None, classDeclarationSyntax.Identifier.Text);
+				context.ReportDiagnostic(diag);
+				continue;
+			}
+
+			if (declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
+			{
+				var nameSpace = declarationSymbol.ContainingNamespace.ToDisplayString();
+				if (nameSpace == "<global namespace>")
+				{
+					var diag = Diagnostic.Create(TextColorToDiagnostic.GlobalNamespace, Location.None, declarationSymbol.Name);
+					context.ReportDiagnostic(diag);
+					continue;
+				}
+
+				var accessModifier = GetClassAccessModifier(declarationSymbol);
+
+				if (accessModifier == string.Empty)
+				{
+					var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidModifierAccess, Location.None, declarationSymbol.Name);
+					context.ReportDiagnostic(diag);
+					continue;
+				}
+
+				textStyleClassList.Add((declarationSymbol.Name, accessModifier, nameSpace));
+			}
+		}
+
+		var options = ((CSharpCompilation)compilation).SyntaxTrees[0].Options as CSharpParseOptions;
+		foreach (var textStyleClass in textStyleClassList)
+		{
+			var textColorToBuilder = @"
+// AutoGenerated Code
+// See: CommunityToolkit.Maui.SourceGenerators.TextColorToGenerator
+
+using System;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.Core.Extensions;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
+
+namespace " + textStyleClass.Namespace + @";
+
+" + textStyleClass.ClassAcessModifier + @" static partial class ColorAnimationExtensions_" + textStyleClass.ClassName + @"
+{
+	/// <summary>
+	/// Animates the TextColor of an <see cref=""Microsoft.Maui.ITextStyle""/> to the given color
+	/// </summary>
+	/// <param name=""element""></param>
+	/// <param name=""color"">The target color to animate the <see cref=""Microsoft.Maui.ITextStyle.TextColor""/> to</param>
+	/// <param name=""rate"">The time, in milliseconds, between the frames of the animation</param>
+	/// <param name=""length"">The duration, in milliseconds, of the animation</param>
+	/// <param name=""easing"">The easing function to be used in the animation</param>
+	/// <returns>Value indicating if the animation completed successfully or not</returns>
+	public static Task<bool> TextColorTo(this " + textStyleClass.Namespace + "." + textStyleClass.ClassName + @" element, Color color, uint rate = 16u, uint length = 250u, Easing? easing = null)
```

Comment:
I might be missing something, but why not use `ITextStyle` as the element type. With that, we don't care about the type of the first parameter, and there is a single generated extension that works for all types that implement `ITextStyle`.

Reply:
Because `ITextStyle` implements the properties that we need as read-only.